src/main/java/com/scottlogic/filters/combine/OrCombineFilters.java

```diff
@@ -4,32 +4,32 @@
 import com.scottlogic.UserPost;
 
 import java.util.ArrayList;
-import java.util.LinkedHashSet;
 import java.util.List;
-import java.util.Set;
 
 public class OrCombineFilters implements PostFilter {
     final private PostFilter filter1;
     final private PostFilter filter2;
 
-    public OrCombineFilters(PostFilter inputFilter1, PostFilter inputFilter2){
+    public OrCombineFilters(PostFilter inputFilter1, PostFilter inputFilter2) {
         filter1 = inputFilter1;
         filter2 = inputFilter2;
     }
 
 
     @Override
     public List<UserPost> filter(List<UserPost> inputList) {
-        if(inputList == null){return null;}
-        List<UserPost> outputList2 = filter1.filter(inputList);
-        List<UserPost> outputList3 = filter2.filter(inputList);
+        if (inputList == null) {
+            return null;
+        }
+        List<UserPost> outputList2 = new ArrayList<>(filter1.filter(inputList));
+        List<UserPost> outputList3 = new ArrayList<>(filter2.filter(inputList));
```

Comment:
So here is a situation where having the `filter` method mutate the `inputList` would have been a really big headache, just thought I'd point it out because of how sensible this code looks to then have been giving a bug.